### PR TITLE
dev: update linter

### PR DESCRIPTION
**Pull request by desimone**
## Summary

Updates golinter-ci. 

- [gofumpt](https://github.com/mvdan/gofumpt) everything
- fix TLS MinVersion to be at least 1.2 on clients
- fix potential decompression bomb in ecjson package
- fix implicit memory aliasing in for loops
- consistent whitespacing

**Checklist**:

- [x] ready for review

### CODE REVIEW

File: internal/encoding/ecjson/ecjson.go [hunk 1]
```diff
@@ -104,8 +110,12 @@ func decompress(data []byte) ([]byte, error) {
 	}
 	defer reader.Close()
 	var buf bytes.Buffer
-	if _, err = io.Copy(&buf, reader); err != nil {
+	n, err := io.CopyN(&buf, reader, maxMemory+1)
```
Comment (desimone) [inline internal/encoding/ecjson/ecjson.go:113]: 👀

File: config/http.go [hunk 2]
```diff
@@ -35,7 +35,8 @@ func (t *httpTransport) update(options *Options) {
 		rootCAs, err := cryptutil.GetCertPool(options.CA, options.CAFile)
 		if err == nil {
 			nt.TLSClientConfig = &tls.Config{
-				RootCAs: rootCAs,
+				RootCAs:    rootCAs,
+				MinVersion: tls.VersionTLS12,
```
Comment (desimone) [inline config/http.go:39]: 👀

File: pkg/grpc/client.go [hunk 3]
```diff
@@ -103,7 +103,7 @@ func NewGRPCClientConn(opts *Options) (*grpc.ClientConn, error) {
 			return nil, err
 		}
 
-		cert := credentials.NewTLS(&tls.Config{RootCAs: rootCAs})
+		cert := credentials.NewTLS(&tls.Config{RootCAs: rootCAs, MinVersion: tls.VersionTLS12})
```
Comment (desimone) [inline pkg/grpc/client.go:106]: 👀

File: pkg/cryptutil/tls.go [hunk 4]
```diff
@@ -47,9 +47,9 @@ func GetCertPool(ca, caFile string) (*x509.CertPool, error) {
 // Finally if there are no matching certificates one will be generated.
 func GetCertificateForDomain(certificates []tls.Certificate, domain string) (*tls.Certificate, error) {
 	// first try a direct name match
-	for _, cert := range certificates {
-		if matchesDomain(&cert, domain) {
-			return &cert, nil
+	for i := range certificates {
```
Comment (desimone) [inline pkg/cryptutil/tls.go:50]: 👀

**Review by calebdoxsey — APPROVED** [review]

**Pull request merged**